IB/ipath - fix prefunit logic in best unit calculation

Broke when multiple HCAs were installed, but not connected, and taskset
was used.

Signed-off-by: John Gregor <john.gregor@qlogic.com>

Index: ofa_kernel-1.4/drivers/infiniband/hw/ipath/ipath_file_ops.c
===================================================================
--- ofa_kernel-1.4.orig/drivers/infiniband/hw/ipath/ipath_file_ops.c	2009-01-13 21:10:03.000000000 -0800
+++ ofa_kernel-1.4/drivers/infiniband/hw/ipath/ipath_file_ops.c	2009-01-13 21:17:37.000000000 -0800
@@ -1748,8 +1748,8 @@ recheck:
 			ipath_dbg("No ports available (none initialized "
 				  "and ready)\n");
 		} else {
-			if (prefunit > 0) {
-				/* if started above 0, retry from 0 */
+			if (prefunit != -1) {
+				/* if had prefunit, retry from 0 */
 				ipath_cdbg(PROC,
 					   "%s[%u] no ports on prefunit "
 					   "%d, clear and re-check\n",
